bootstrap

```diff
@@ -1,4 +1,4 @@
-#! /bin/sh
+#!/usr/bin/env bash
```

Comment:
We're not going to change scripts to hard depend on `bash`. I would suggest keeping the shebang as is, and add variables for `AUTORECONF` and `AUTOMAKE` that default to `autoreconf` and `automake` respectively. Then, when calling `./bootstrap`, one can pass `AUTORECONF=true` to skip the step.

Reply:
Oh sorry. This was just for testing. I can revert it

Reply:
got it, good idea

Reply:
Yikes, I forgot this. Sorry. Will do it tomorrow @mintsuki

Reply:
Addressed. I hope the solution is fine to you?

Reply:
There are issues with this, imho. I will shortly make a commit addressing the issue as I originally stated in this thread.

Reply:
Okay, I pushed the aforementioned change. Now you can do `AUTOMAKE=true AUTOCONF=true ./bootstrap` and bootstrap will skip the autotools steps.

Likewise, if the submodules are already cloned, running `./bootstrap` will not download them again, and it will only run the autotools steps. This should remove the need to manually invoke `autoreconf`.

Reply:
I rebased the branch. Thanks!

(IMHO your solution sounds unusual to me. Is this something that is widely used? Because `AUTOMAKE=true` referring to the `true` binary/bash builtin is far from intuitive. One reads it as a boolean instead.)

Reply:
Yeah it's a pretty standard idiom.